gobblin-data-management/src/main/java/gobblin/data/management/copy/CopyableFile.java

```
@@ -68,6 +78,83 @@
+      return this;
+    }
+
+    public CopyableFile build() throws IOException {
+
+      if(this.destinationOwnerAndPermission == null) {
+        this.destinationOwnerAndPermission =
+            new OwnerAndPermission(this.origin.getOwner(), this.origin.getGroup(), this.origin.getPermission());
+      }
+      if(this.ancestorsOwnerAndPermission == null) {
+        this.ancestorsOwnerAndPermission = replicateOwnerAndPermission(this.originFs, this.origin.getPath());
+      }
+      if(this.checksum == null) {
+        FileChecksum checksumTmp = this.originFs.getFileChecksum(origin.getPath());
+        this.checksum = checksumTmp == null ? new byte[0] : checksumTmp.getBytes();
+      }
+
+      return new CopyableFile(origin, destination, relativeDestination, destinationOwnerAndPermission,
+          ancestorsOwnerAndPermission, null);
+    }
+
+    private List<OwnerAndPermission> replicateOwnerAndPermission(final FileSystem originFs, final Path path) {
+
+      Path rootPathTmp = PathUtils.getPathWithoutSchemeAndAuthority(this.rootPath);
+
+      List<OwnerAndPermission> ancestorOwnerAndPermissions = Lists.newArrayList();
+      try {
+        Path currentPath = PathUtils.getPathWithoutSchemeAndAuthority(path);
+        while (currentPath != null && currentPath.getParent() != null && !currentPath.getParent().equals(rootPathTmp)) {
+          currentPath = currentPath.getParent();
+          final Path thisPath = currentPath;
+          ancestorOwnerAndPermissions.add(ownerAndPermissionCache().get(originFs.makeQualified(currentPath),
+              new Callable<OwnerAndPermission>() {
+            @Override public OwnerAndPermission call() throws Exception {
+              FileStatus fs = originFs.getFileStatus(thisPath);
+              return new OwnerAndPermission(fs.getOwner(), fs.getGroup(), fs.getPermission());
+            }
+          }));
+        }
+      } catch (ExecutionException ee) {
+        // When cache loader failed.
```

Better to take some action instead of swallowing the exception.